Modules/Tokens/src/Tokens.Infrastructure/Persistence/Repository/TokensRepository.cs

```diff
@@ -24,28 +24,29 @@ public TokensRepository(TokensDbContext dbContext)
         _readonlyTokensDbSet = dbContext.Tokens.AsNoTracking();
     }
 
-    public async Task<Token> Find(TokenId id)
+    public async Task<Token> Find(TokenId id, IdentityAddress? address)
     {
         var getMetadata = _readonlyTokensDbSet
             .Where(Token.IsNotExpired)
+            .Where(Token.CanBeCollectedBy(address))
             .FirstWithId(id);
 
         var token = await getMetadata ?? throw new NotFoundException(nameof(Token));
 
         return token;
     }
 
-    public async Task<DbPaginationResult<Token>> FindAllWithIds(IEnumerable<TokenId> ids, PaginationFilter paginationFilter, CancellationToken cancellationToken)
+    public async Task<DbPaginationResult<Token>> FindAllWithIds(IdentityAddress createdForOrBy, IEnumerable<TokenId> ids, PaginationFilter paginationFilter, CancellationToken cancellationToken)
     {
-        return await Find(null, ids, paginationFilter, cancellationToken);
+        return await Find(null, createdForOrBy, ids, paginationFilter, cancellationToken);
     }
 
     public async Task<DbPaginationResult<Token>> FindAllOfOwner(IdentityAddress owner, PaginationFilter paginationFilter, CancellationToken cancellationToken)
     {
-        return await Find(owner, Array.Empty<TokenId>(), paginationFilter, cancellationToken);
+        return await Find(owner, null, [], paginationFilter, cancellationToken);
     }
 
-    private async Task<DbPaginationResult<Token>> Find(IdentityAddress? owner, IEnumerable<TokenId> ids, PaginationFilter paginationFilter, CancellationToken cancellationToken)
+    private async Task<DbPaginationResult<Token>> Find(IdentityAddress? owner, IdentityAddress? createdForOrBy, IEnumerable<TokenId> ids, PaginationFilter paginationFilter, CancellationToken cancellationToken)
```

Comment:
Why are there two parameters for an address, where the first is can be only the owner, and the second can be the owner OR the `createdFor`? I don't think that this makes any sense. Am I missing something?

Reply:
changes made